### PR TITLE
feat: add error fallback component

### PR DESCRIPTION
**Pull request by saihaj**
## Summary of PR
Modifies the desktop component and implements it here

image

### Tests for unexpected behavior
Forced it to break and then clearing it makes it work.

### Time spent on PR
1 hour

**Preview**: https://shabad-os-theme-tool-git-error-fallback.saihaj.vercel.app
### Reviewers
@Harjot1Singh @bhajneet

### CODE REVIEW

**Comment by bhajneet** [comment]
If it crashes with non-default local storage, can't it simply clear local
storage automatically to try and fix it?

**Comment by saihaj** [comment]
> If it crashes with non-default local storage, can't it simply clear local storage automatically to try and fix it?
> […](#)

Most probably with non-default local storage it won't crash, unless someone forces changes. Right now it can only reset itself if you try to delete properties. 

This component is just as a safety measure so that if in future there are any breaking changes then the old version of app can sync up with new one instead of showing a blank screen.

File: src/components/ErrorFallback.css [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    color: #1A191A;
+    font-size: 1.15em;
+    margin: 0;
+  }
+  
+  .error-fallback .buttons button {
+    margin: 20px;
+  }
+  
+  .error-fallback .error {
```
Comment (harjot1singh) [inline src/components/ErrorFallback.css]: Your formatting is broke in this file. Please comb through it to check spacing is correct (2 per indent)

File: src/components/ErrorFallback.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+export const withErrorFallback = Comp => class WithErrorFallback extends Component {
+  state = { error: null }
+
+  static getDerivedStateFromError( error ) {
+    return { error }
+  }
+
+  render() {
+    const { error } = this.state
+    if ( error ) {
```
Comment (harjot1singh) [inline src/components/ErrorFallback.js:93]: newline

File: src/components/ErrorFallback.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+            <h5>{`ShabadOS Theme Tool v${version}`}</h5>
+
+          </div>
+
+        </>
+      )
+    }
+
+
```
Comment (harjot1singh) [inline src/components/ErrorFallback.js]: extra newline

File: src/components/ErrorFallback.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
+import React, { Component, useState, useEffect } from 'react'
+import { string, bool } from 'prop-types'
+
+import { Typography, Button, Grid } from '@material-ui/core'
+
+import './ErrorFallback.css'
```
Comment (harjot1singh) [inline src/components/ErrorFallback.js]: The closest import goes at the end

File: src/components/ErrorFallback.css [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+.error-fallback-body .logo {
+  height: 30vmin;
+}
+
+
+.error-fallback-body h5 {
+  color: white;
+  font-size: 2em;
+}
```
Comment (harjot1singh) [inline src/components/ErrorFallback.css]: Missing EOF newline

**Review by harjot1singh — CHANGES_REQUESTED** [review]
I would suggest adding some margin from the top or similar.

**Comment by saihaj** [comment]
> I would suggest adding some margin from the top or similar.

Move down the popup ?

**Review by harjot1singh — APPROVED** [review]

**Comment by harjot1singh** [comment]
> > I would suggest adding some margin from the top or similar.
> 
> Move down the popup ?

Yes

**Review by harjot1singh — CHANGES_REQUESTED** [review]
Move it down a tad

**Review by harjot1singh — APPROVED** [review]
All good.

I think in the future, we should start thinking about bundling up our own component library/UI presets etc.

**Pull request merged**